tests/deploy/test_scheduler_agent.py

```diff
@@ -1232,6 +1215,67 @@ def make_resources(
     assert agent.executor_manager.executors["agent2"].execute_count == 3
     assert agent.executor_manager.executors["agent3"].execute_count == 0
 
+    #########################
+    # Verify failure events #
+    #########################
+    agent.executor_manager.reset_executor_counters()
+    # make agent1's executor also managed to make r1 fail and succeed as we please
+    executor1: ManagedExecutor = ManagedExecutor()
+    agent.executor_manager.register_managed_executor("agent1", executor1)
+
+    # release change to r1, and make its deploy fail
+    resources = make_resources(r1_value=2, r2_value=1, r3_value=0)
```

Comment:
I'd keep the change restricted to what you actually want to change
```suggestion
    resources = make_resources(r1_value=6, r2_value=0, r3_value=0)
```
(previous "export" was 5, 0, 0).

I admit this incremental approach is not great. At some point I tried to make each of the scenarios more standalone like I did with the resetting of the counters, but it didn't work out.